zerver/tests/test_message_fetch.py

```diff
@@ -219,6 +220,86 @@ def test_add_term_using_channels_operator_and_public_operand_negated(self) -> No
+        term = NarrowParameter(operator="channels", operand="archived")
+        self._do_add_term_test(
+            term,
+            "WHERE recipient_id IN (__[POSTCOMPILE_recipient_id_1])",
+        )
+
+        # Add new channels
+        channel_dicts: list[StreamDict] = [
+            {
+                "name": "public-channel",
+                "description": "Public channel with public history",
+            },
+            {
+                "name": "private-channel",
+                "description": "Private channel with non-public history",
+                "invite_only": True,
+            },
+            {
+                "name": "private-channel-with-history",
+                "description": "Private channel with public history",
+                "invite_only": True,
+                "history_public_to_subscribers": True,
+            },
+        ]
+        realm = get_realm("zulip")
+        created, existing = create_streams_if_needed(realm, channel_dicts)
+        self.assert_length(created, 3)
+        self.assert_length(existing, 0)
+
+        # Archive the streams in the created list
+        for archived_stream in created:
+            do_deactivate_stream(archived_stream, acting_user=None)
+
+        # Verify the term behavior after archiving streams
+        self._do_add_term_test(
+            term,
+            "WHERE recipient_id IN (__[POSTCOMPILE_recipient_id_1])",
+        )
+
+    def test_add_term_using_channels_operator_and_archived_operand_negated(self) -> None:
+        term = NarrowParameter(operator="channels", operand="archived", negated=True)
+        self._do_add_term_test(
+            term,
+            "WHERE (recipient_id NOT IN (__[POSTCOMPILE_recipient_id_1]))",
+        )
+
+        # Add new channels
+        channel_dicts: list[StreamDict] = [
+            {
+                "name": "public-channel",
+                "description": "Public channel with public history",
+            },
+            {
+                "name": "private-channel",
+                "description": "Private channel with non-public history",
+                "invite_only": True,
+            },
+            {
+                "name": "private-channel-with-history",
+                "description": "Private channel with public history",
+                "invite_only": True,
+                "history_public_to_subscribers": True,
+            },
+        ]
+        realm = get_realm("zulip")
+        created, existing = create_streams_if_needed(realm, channel_dicts)
+        self.assert_length(created, 3)
+        self.assert_length(existing, 0)
+
+        # Archive the streams in the created list
+        for archived_stream in created:
+            do_deactivate_stream(archived_stream, acting_user=None)
+
+        # Verify the term behavior with negation after archiving streams
+        self._do_add_term_test(
+            term,
+            "WHERE (recipient_id NOT IN (__[POSTCOMPILE_recipient_id_1]))",
+        )
```

Comment:
These tests seems to do a lot of setup without much in the way of assertions. It seems like you could at the very least test both the normal operators and negation without creating all the channels all over again in a single test.